profiles/api.py

```diff
@@ -87,15 +93,15 @@ def get_queryset(self):
         return self.queryset.filter(user=self.request.user)
 
     def get_object(self):
-        try:
-            profile = self.get_queryset().get(user=self.request.user)
-        except Profile.DoesNotExist:
-            # TODO: create profiles when the user is created, not here.
-            profile = Profile.objects.create(user=self.request.user)
-        except (TypeError, ValueError):
-            raise Http404
+        return self.get_queryset().filter(user=self.request.user)
```

Comment:
Does this actually work? Shouldn't `get_object()` return an object instance instead of a queryset?

Reply:
Hm, it works, but yeah, you would think that it needs to return an object...